### ISSUE TITLE
Volume blue halo with "White" RGB vs constant SPECTRUM.

### ISSUE THREAD

**Issue by webanck** [mmp/pbrt-v4#28]
Hello and thanks for sharing this new version of pbrt; I am very happy to discover the new features, especially the participating media ones!

When rendering a volumetric scene with "white" RGB values instead of a constant spectrum such as in the [bunny-cloud](https://github.com/mmp/pbrt-v4-scenes/tree/master/bunny-cloud/), a blue halo is produced.

The provided `"spectrum sigma_s" [200 10 900 10] "spectrum sigma_a" [200 .5 900 .5]`:
![bunny-cloud-spectrum](https://user-images.githubusercontent.com/6747237/91988253-585da780-ed2f-11ea-9fbb-b2321a784b0e.png)
Versus `"rgb sigma_s" [10 10 10] "rgb sigma_a" [.5 .5 .5]`:
![bunny-cloud-rgb](https://user-images.githubusercontent.com/6747237/91988250-57c51100-ed2f-11ea-94a4-79d32ee3079c.png)

Is it the expected behavior, knowing that the automatic converter between pbrt-v3 and pbrt-v4 scenes keeps the RGB values, even when it is "white" ?

**Comment by shadeops** [comment]
I don't think this has to do with the conversion, for example rendering a cloudy torus with rgb values gives an interesting faint blue grid structure.

![cloud](https://user-images.githubusercontent.com/2104033/92082859-f8d8b980-ed79-11ea-84ac-69017be200b3.png)

This looks similar to what surrounds the bunny, you can see some of the structure in the ears especially.

[cloud.zip](https://github.com/mmp/pbrt-v4/files/5166832/cloud.zip)
Rendered with https://github.com/mmp/pbrt-v4/tree/ca0db69

**Comment by mmp** [comment]
(Sorry I haven't had a chance to look into this yet, but yes, it doesn't look right..)

Possibly related, via @anderslanglands: the attached scene is a big white diffuse quad, lit by a white light. If the reflectance is 0.999, the average RGB channel values are all just under one, as would be expected
```
	                       B: min      0.98291 max      1.01758 avg     0.999213 (0 infinite, 0 not-a-number)
	                       G: min     0.994629 max      1.00098 avg     0.999048 (0 infinite, 0 not-a-number)
	                       R: min     0.992676 max      1.00586 avg     0.998826 (0 infinite, 0 not-a-number)
```

If the diffuse reflectance is 1, then things go blue-ish:
```
	                       B: min      0.98584 max      1.02051 avg        1.002 (0 infinite, 0 not-a-number)
	                       G: min     0.975098 max     0.981445 avg     0.979393 (0 infinite, 0 not-a-number)
	                       R: min     0.958984 max     0.971191 avg     0.964616 (0 infinite, 0 not-a-number)
```
[x.pbrt.txt](https://github.com/mmp/pbrt-v4/files/5177321/x.pbrt.txt)

**Comment by mmp** [comment]
The issue here is that the RGB values for sigma_a/sigma_s end up being represented using RGBSpectrum, which multiplies the value it returns by color space's illuminant. What we want here is RGBReflectanceSpectrum, which doesn't do that.

*But*, RGBReflectanceSpectrum only handles values in [0,1] (i.e., reflectances). Plus, it's clearly misnamed then.

So the proper fix will probably be to rename RGBReflectanceSpectrum to RGBSpectrum and fix it to handle values outside of [0,1], rename the old RGBSpectrum to RGBIlluminantSpectrum, and to patch everything up accordingly. This should be straightforward; I'll try to get to it soon.

**Comment by shadeops** [comment]
This is looking good when rendered with https://github.com/mmp/pbrt-v4/tree/0e4b5ae

![cloud](https://user-images.githubusercontent.com/2104033/95156714-0eac1680-074c-11eb-8b07-def24180dcd8.jpg)

Thanks Matt!

**Comment by mmp** [comment]
(Are you sure that one was rendered with RGB? I was still seeing a slight blue tinge with that cloudy doughnut scene that is now fixed(*) with e6b4d90. Here is ToT now:

![image](https://user-images.githubusercontent.com/102567/95230799-2cfb2c00-07b7-11eb-8ed7-5be5a483ff18.png)

(*) RGB of exactly 1,1,1 still comes out blue-ish; I need to chase that down but think I will file that as a separate bug. The attachment is rendered with 0.99 0.99 0.99. I've also confirmed that bunny-cloud comes out correctly with those RGB values.

Thanks for reporting this!

**Comment by shadeops** [comment]
Opps sorry Matt, that was my bad I should have been more clear in what I was doing. My render in https://github.com/mmp/pbrt-v4/issues/28#issuecomment-704010808 was with a sigma_a|s of 0.999 as I was just testing the fix for majorant grid artifacts.

Thanks for making the Issue #54
